### PR TITLE
[NCL-3568] add option to override version suffix in lookup report

### PR DESCRIPTION
**Pull request by janinko**

### CODE REVIEW

File: reports-backend/src/main/java/org/jboss/da/products/impl/AggregatedProductProvider.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
  *
  * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
  */
-@ApplicationScoped
+//@ApplicationScoped // Should be Request scoped
```
Comment (thescouser89) [inline reports-backend/src/main/java/org/jboss/da/products/impl/AggregatedProductProvider.java]: Stupid question: does injection work when there are no annotation on the class?

File: common/src/main/java/org/jboss/da/common/version/VersionParser.java [hunk 2]
```diff
@@ -9,10 +9,23 @@
 @ApplicationScoped
 public class VersionParser {
 
-    private static final String PATTERN_SUFFIX_BUILT_VERSION = "[.-]redhat-(\\d+)";
+    private static final String DEFUALT_SUFFIX = "redhat";
```
Comment (thescouser89) [inline common/src/main/java/org/jboss/da/common/version/VersionParser.java]: Typo on DEFAULT

File: reports-backend/src/main/java/org/jboss/da/products/impl/AggregatedProductProvider.java (hunk 1 above)
Comment (janinko) [reply reports-backend/src/main/java/org/jboss/da/products/impl/AggregatedProductProvider.java]: Yup. The default scope is [Dependent](https://docs.jboss.org/cdi/api/1.0/javax/enterprise/context/Dependent.html), which works here now, but the Request scope would be better. However Request scope doesn't work with websockets.

Comment (thescouser89) [reply reports-backend/src/main/java/org/jboss/da/products/impl/AggregatedProductProvider.java]: Alright! I haven't found any doc on that, but I guess it's not even documented(?)

**Comment by thescouser89** [comment]
LGTM (minus tiny typo)! 👍

File: reports-backend/src/main/java/org/jboss/da/products/impl/AggregatedProductProvider.java (hunk 1 above)
Comment (janinko) [reply reports-backend/src/main/java/org/jboss/da/products/impl/AggregatedProductProvider.java]: https://docs.oracle.com/javaee/6/tutorial/doc/gjbbk.html
> Dependent
> @Dependent
> The default scope if none is specified; it means that an object exists to serve exactly one client (bean) and has the same lifecycle as that client (bean).

**Pull request merged**